packages/react/src/transactions/useFollow.ts

```diff
@@ -97,35 +92,36 @@ export function useFollow({ profile }: UseFollowArgs) {
     | WalletConnectionError
     | null
   >(null);
+  const [isPending, setIsPending] = useState<boolean>(false);
   const follow = useFollowController();
   const activeWallet = useActiveWallet();
   const { data: activeProfile } = useActiveProfile();
```

Comment:
interesting observation that `useActiveWallet` and `useActiveProfile` do not follow the same approach of returning `wallet/profile`

Reply:
It is interesting, both are using reactive vars but active profile has this logic:

```typescript
if (state === ApplicationsState.LOADING) {
    return {
      loading: true,
      data: undefined,
    };
  }
  ```
  
  do we know the reason for the discrepancy?  @cesarenaldi @desfero

Reply:
it appears as a precaution to not start loading queries until we know that the app signaled readiness.
I can see that when the code evolves here (https://github.com/lens-protocol/lens-sdk/blob/3556929122e89389bf9d4640afd03810a0516065/packages/domain/src/use-cases/lifecycle/Bootstrap.ts#L70-L77) we can end up with some race condition between a call to `activeProfile.loadActiveProfileByOwnerAddress` and `applicationPresenter.signalReady.

Reply:
after line 73 in the code above, we know for sure that we have a an active profile, so it looks like the checks are fine?

Reply:
i guess soo, maybe @cesarenaldi knows some edge cases?

Reply:
It has to do with avoiding flash of logged-out content. To know if we have done-done and if we have a profile or not the only reliable way was to wait for the ready signal.

Arguably we can shift the logged-in/out responsibility to the `useActiveWallet` and let the `useActiveProfile` just propagate the content of the Reactive Var, thoughts?

Reply:
Continuing this conversation here: https://avara.height.app/T-3548 🫡